crypto/s2n_pkey.c

```diff
@@ -0,0 +1,136 @@
+#include <openssl/evp.h>
+#include <openssl/x509.h>
+
+#include "error/s2n_errno.h"
+
+#include "crypto/s2n_pkey.h"
+
+#include "utils/s2n_safety.h"
+
+int s2n_asn1der_to_private_key(struct s2n_pkey *priv_key, struct s2n_blob *asn1der)
+{
+    int ret;
```

Comment:
nit: This is a bit more readable to me if we move this declaration down to where it's first used.